src/storage/mod.rs

```diff
@@ -1515,6 +1519,76 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
         Ok(())
     }
 
+    pub fn flashback_to_version<R: RaftStoreRouter<E::Local> + 'static>(
+        &self,
+        mut req: FlashbackToVersionRequest,
+        raft_router: R,
+        callback: Callback<()>,
+    ) -> Result<()> {
+        let ctx = req.take_context();
+        Self::check_api_version_ranges(
+            self.api_version,
+            ctx.api_version,
+            CommandKind::flashback_to_version,
+            [(Some(req.get_start_key()), Some(req.get_end_key()))],
+        )?;
+
+        let region_id = ctx.get_region_id();
+        // Reject any following scheduling, read or writing request, and wait for the
+        // applied index to reach the latest committed index.
+        let (tx, rx) = sync_channel(1);
```

Comment:
the future is spawned on grpc thread, you must't do sync wait here, otherwise it blocks the grpc thread.